SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestTearOff/SpatialTestTearOff.cpp

```diff
@@ -0,0 +1,186 @@
+			RegisterAutoDestroyActor(SpawnedTearOffActor);
+
+			// Make sure the Actor was correctly spawned
+			if (IsValid(SpawnedTearOffActor))
+			{
+				FinishStep();
+			}
+		});
+
+	// Allow for potential replication to propagate before checking that the SpawnedTearOffActor was not received.
+	AddStepFromDefinition(WorkerWaitForTimeStepDefinition, FWorkerDefinition::AllWorkers);
+
+	// Clients check that they have not received the newly spawned Actor
+	AddStep(TEXT("SpatialTestTearOffUniversalSpawnedActorReferenceSetup"), FWorkerDefinition::AllClients, nullptr, [this]()
+		{
+			TArray<AActor*> ReplicatedTearOffActors;
+			UGameplayStatics::GetAllActorsOfClass(GetWorld(), AReplicatedTearOffActor::StaticClass(), ReplicatedTearOffActors);
+
+			// Only the StartupTearOffActor is expected to exist in the world at this point.
+			if (ReplicatedTearOffActors.Num() == 1)
+			{
+				FinishStep();
+			}
+		});
+
+	// Spawn a replicated Actor that does not call TearOff on BeginPlay().
+	AddStep(TEXT("SpatialTestTearOffServerSpawnReplicatedActor"), FWorkerDefinition::Server(1), nullptr, [this]()
+		{
+			SpawnedReplicatedActorBase = GetWorld()->SpawnActor<AReplicatedTestActorBase>(SpwanedReplicatedActorBaseLocation, FRotator::ZeroRotator, FActorSpawnParameters());
+			RegisterAutoDestroyActor(SpawnedReplicatedActorBase);
+			FinishStep();
+		});
+
+	// Make sure the spawned actor is correctly replicated and set a reference to it.
+	AddStep(TEXT("SpatialTestTearOffClientsSpawnedActorReferenceSetup"), FWorkerDefinition::AllClients, nullptr, nullptr, [this](float DeltaTime)
+		{
+			TArray<AActor*> ReplicatedActors;
+			UGameplayStatics::GetAllActorsOfClass(GetWorld(), AReplicatedTestActorBase::StaticClass(), ReplicatedActors);
+
+			// Since ReplicatedTearOffActor inherits from AReplicatedTestActorBase, 2 AReplicatedTestActorBase are expected at this point:
+			// The start-up ReplicatedTearOfActor, and the newly spawned AReplicatedTestActorBase object.
+			if (ReplicatedActors.Num() == 2)
+			{
+				// Ensure the reference to the correct Actor is set.
+				if (ReplicatedActors[0]->IsA(AReplicatedTearOffActor::StaticClass()))
+				{
+					SpawnedReplicatedActorBase = Cast<AReplicatedTestActorBase>(ReplicatedActors[1]);
+				}
+				else
+				{
+					SpawnedReplicatedActorBase = Cast<AReplicatedTestActorBase>(ReplicatedActors[0]);
+				}
+
+				if (IsValid(SpawnedReplicatedActorBase) && SpawnedReplicatedActorBase->GetActorLocation().Equals(SpwanedReplicatedActorBaseLocation, 1.0f))
+				{
+					FinishStep();
+				}
+			}
+		}, 5.0f);
+
+	// Tear off the spawned replicated Actor and move it to a new location.
+	AddStep(TEXT("SpatialTestTearOffServerSetTearOffAndMoveActor"), FWorkerDefinition::Server(1), nullptr, [this]()
+		{
+			SpawnedReplicatedActorBase->TearOff();
+			SpawnedReplicatedActorBase->SetActorLocation(FVector(30.0f, -20.0f, 80.0f));
```

Comment:
Does this not propagate in native? I would have thought that since you mentioned native has different semantics that this would pose a problem in native.

Reply:
I'm not sure I am following what you mean, but the difference that I have noticed with Native is that if you have a Startup Actor that calls TearOff on BeginPlay, then it takes longer to actually prevent replication than in Spatial (did not dig too much into this to see why this happens). In this case however I didn't notice any difference.

Reply:
Okay, I just thought if native takes some time to actually tear off in begin play, that the order of (TearOff) (ReplicateProperty) in the same tick would also not be guaranteed. But if it works, then it works.

Reply:
Out of curiosity, I would want to take a look at the BeginPlay case, but I am not sure what's a good approach for this, can you think of a way of checking/ a starting point on  why the tear off in BeginPlay takes a bit longer?

Reply:
Would be kind of annoying - would have to check the native replication flow and see where bTearOff is checked and why it would not be respected / alternatively if a replication tick happens before BeginPlay (but I believe that shouldn't be the case).